lib/IRGen/GenMeta.cpp

```diff
@@ -3858,6 +3869,9 @@ namespace {
     // Visitor methods.
 
     void addValueWitnessTable() {
+      if (IGM.getOptions().LazyInitializeClassMetadata)
+        return B.addNullPointer(IGM.WitnessTablePtrTy);
```

Comment:
I think the right way of thinking about this is: if (1) the Swift runtime has a default VWT that it fills in during allocation for foreign class metadata and (2) the VWT we need is that default VWT, then we can use a null pointer here instead of the address of the VWT.

(1) will be true on all targets *except* Apple platforms with a deployment target <= the Swift 5 targets.  We may need to build the abstractions to make it easy to query this, but they will be useful for a lot more than this.  See `ObjCRuntime` in Clang for what this should basically look like.

(2) will always be true because we don't have foreign classes that use weird reference-counting right now, but we should at least leave a comment to explain why we can be unconditional.

Reply:
Hmm, I think that it is beneficial to get this change merged and address 1 in a follow up change.  This change is needed to enable the nightlies (and others) to start building Foundation for Windows.

Reply:
Something like `hasDefaultForeignClassValueWitnesses()`.

I'm fine with doing the SwiftRuntime work in a follow-up.

Reply:
Typo: `clss` for `class`.  Also, the comment needs to reflect that this is dependent on the runtime version, e.g.:

```
As of Swift 5.1, the runtime will fill in a default VWT during allocation of foreign
class metadata.  We rely on this for correctness on COFF, where we can't necessarily
reference the standard VWT from the metadata candidate, but it's a good optimization
everywhere.  FIXME: take advantage of this on other targets when targeting a
sufficiently recent runtime.

The default VWT uses ObjC-compatible reference counting if ObjC interop is enabled
and Swift-compatible reference counting otherwise.  That's currently always good
enough for foreign classes, so we can unconditionally rely on the default VWT.
```

Also, you should be able to add an assertion about the reference counting used by `Target`.